packages/lastrev-integration-contentful/src/contentLoader/contentLoader.ts

```diff
@@ -0,0 +1,186 @@
+import { get, map } from 'lodash/fp';
+import Adapter, { AdapterConfig } from '@last-rev/adapter-contentful';
+import { ContentfulClientApi } from 'contentful';
+import { flatten, groupBy } from 'lodash';
+import compose from './compose';
+import { createLoader } from './createLoader';
+import fetchEntry from './fetchEntry';
+
+interface ContentLoaderConfig {
+  client: ContentfulClientApi;
+  composers?: any;
+  syncAllEntriesForContentType: any;
+  transform?: (x: unknown) => unknown;
+}
+const DEFAULT_CONFIG = {
+  useSyncAPI: process.env.USE_SYNC_API || false,
+  useFileCache: process.env.USE_FILE_CACHE || false
+};
+
+interface Key {
+  contentTypeId?: string;
+  id?: string;
+  slug?: string;
+  locale?: string;
+}
+const logger = (...args: any) => console.log('ContentLoader', ...args);
+// Produces the same key applied to load and when parsing an Entry
+const getKey = ({ locale, contentTypeId, slug, id }: Key) => (id ? { locale, id } : { locale, contentTypeId, slug });
+const getKeyString = (x: any) => JSON.stringify(getKey(x));
+
+const resolveSettled = (promises) =>
+  promises.map((p) => {
+    if (p?.status === 'rejected') {
+      console.log('Error', p?.reason);
+    }
+    return p?.value || null;
+  });
+const fetchEntries = async ({
+  client,
+  contentTypeId,
+  keys,
+  useSyncAPI,
+  useFileCache,
+  syncAllEntriesForContentType,
+  contentJsonDirectory
+}) => {
+  let entries = [];
+  if (useFileCache) {
+    console.log('useFileCache');
+    // Dynamic import to prevent fs dependency on client browsers
```

Comment:
How does this work? Is it mostly just to prevent the dependency, but if you try to call it from the client side it will still fail?